logstash-core/lib/logstash/agent.rb

```diff
@@ -260,16 +261,17 @@ def create_pipeline(settings, config=nil)
     end
   end
 
-  def fetch_config(settings)
-    @config_loader.format_config(settings.get("path.config"), settings.get("config.string"))
+  def fetch_config
+    # TODO(ph) multiple pipeline, we only support one config for now
+    @config_loader.pipeline_config
   end
 
   # since this method modifies the @pipelines hash it is
   # wrapped in @upgrade_mutex in the parent call `reload_state!`
   def reload_pipeline!(id)
     old_pipeline = @pipelines[id]
-    new_config = fetch_config(old_pipeline.settings)
-    if old_pipeline.config_str == new_config
+    new_config = fetch_config
```

Comment:
Yes that's  the idea i've discussed about diffing the current running pipeline vs the new available configs, but I wanted to limit the changes in the `reload_pipeline!`  to not impact too much your work on the multiple pipeline so I have opted to use a similar interface.